src/webview-contract/messaging.ts

```diff
@@ -6,12 +6,30 @@ export type MessageDefinition = {
     [commandName: string]: any
 };
 
+/**
+ * Any object containing all the keys (commands) for a message definition.
+ */
+export type CommandKeys<TMsgDef> = {
+    [P in Command<TMsgDef>]: any
+};
+
 /**
  * A component that messages can be sent to. Both Webviews and the
  * VS Code extension can act as `MessageSink` instances.
  */
-export interface MessageSink<TPostMsgDef extends MessageDefinition> {
-    postMessage(message: Message<TPostMsgDef>): void
+export type MessageSink<TMsgDef extends MessageDefinition> = {
```

Comment:
This is the change that enables posting messages with simpler syntax:
- Before: `postMessage({ command: "customMessage", parameters: args })`
- After: `postCustomMessage(args)`

The new `asMessageSink` function converts implementations of the former to the latter.